queries.sql

```diff
@@ -1,10 +1,65 @@
 /*Queries that provide answers to the questions from all projects.*/
 
+/* Complete the queries requested in (Milestone 1) */
 SELECT * FROM animals WHERE name LIKE '%mon';
 SELECT name FROM animals WHERE date_of_birth BETWEEN '2016-01-01' AND '2019-01-01';
 SELECT name FROM animals WHERE neutered = 'true' AND escape_attempts < 3;
 SELECT date_of_birth FROM animals WHERE name IN ( 'Agumon', 'Pikachu' ); 
 SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = 'true';
 SELECT * FROM animals WHERE name NOT LIKE 'Gabumon';
-SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+/* Complete the transactions and quesries requested in (Milestone 2) */
+
+/* Update species column transaction and rollback */
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
+
+/* Update scpecies column completed transaction */
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
+
+/* Delete all records from animals table and rollback */
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+
+/* Transaction with savepoint */
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT SP1;
+UPDATE animals SET weight_kg = weight_kg*1;
+ROLLBACK TO SP1;
+UPDATE animals SET weight_kg = weight_kg*-1 WHERE weight_kg < 0;
+COMMIT;
+
```

Comment:
- [ ] <b>REQUIRED:</b> Kindly add query that verifies that changes are made on each before and after `COMMIT` and `ROLLBACK` command. something like this:

```sql
   BEGIN; -- start transaction

   UPDATE animals
   SET species = 'unspecified'; -- make change

   SELECT species from animals; -- verify that change was made
   ROLLBACK;

   SELECT species from animals; -- verify that change was undone